### PR TITLE
Improve regexps

### PR DESCRIPTION
**Pull request by rafatower**
@Kartones @juanignaciosl please review

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3612/))

File: app/models/table.rb [hunk 1]
```diff
@@ -932,7 +936,11 @@ def update_row!(row_id, raw_attributes)
             invalid_column = attributes.invert[invalid_value] # which is the column of the name that raises error
             new_column_type = get_new_column_type(invalid_column)
             if new_column_type
-              user_database.set_column_type self.name, invalid_column.to_sym, new_column_type
+              user_database.set_column_type self.name, invalid_column.to_sym, new_column_typ
```
Comment (rafatower) [inline app/models/table.rb]: let me fix that typo :(

**Comment by juanignaciosl** [comment]
:+1:

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3613/))

File: app/controllers/api/json/oembed_controller.rb [hunk 2]
```diff
@@ -16,13 +16,14 @@ def show
...
-    raise ActionController::RoutingError.new('Incorrect height') if (height =~ /^[0-9]+(%|px)?$/).nil?
+    raise ActionController::RoutingError.new('Incorrect width') if (width =~ /\A[0-9]+(%|px)?\z/).nil?
+    raise ActionController::RoutingError.new('Incorrect height') if (height =~ /\A[0-9]+(%|px)?\z/).nil?
 
     uri = URI.parse(url)
 
     begin
-      uuid = /(\w{8}-\w{4}-\w{4}-\w{4}-\w{12})/.match(uri.path)[0]
+      # TODO: remove this method and use  app/helpers/carto/uuidhelper.rb. Not used yet because this changed was pushed before
+      uuid = /\A(\w{8}-\w{4}-\w{4}-\w{4}-\w{12})\z/.match(uri.path)[0]
```
Comment (Kartones) [inline app/controllers/api/json/oembed_controller.rb]: Could you instead use
```
 def is_uuid?(text)
        !(Regexp.new(%r{\A#{UUIDTools::UUID_REGEXP}\Z}) =~ text).nil?
      end
```
until the proper helper is in place

**Comment by Kartones** [comment]
:+1:

File: app/controllers/api/json/oembed_controller.rb (hunk 2 above)
Comment (rafatower) [inline app/controllers/api/json/oembed_controller.rb]: sure

Comment (rafatower) [inline app/controllers/api/json/oembed_controller.rb]: After checking with Daniel (who authored those lines) I shouldn't check strictly that the `uri.path` _is_ a `uuid` but that it _contains_ a `uuid`. Subtle distinction that can break things :)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3625/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3631/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3633/))

**Pull request merged**